### PR TITLE
fix: bug fixes in resource policies

### PR DESCRIPTION
**Pull request by praneetap**
Added support for intrinsics Ref for api auth
Added support for fn::if in CustomStatements
Fixed the policy getting created for vpc whitelist to be 'OR' instead of 'AND'

*Issue #, if available:*
#1249 #1372 #1218 
*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [ ] Write/update tests
- [ ] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/intrinsics.py [hunk 1]
```diff
@@ -146,6 +146,23 @@ def is_instrinsic(input):
...
+    key that is the name of the intrinsics.
+
+    :param input: Input value to check if it is an intrinsic (all except Ref)
+    :return: True, if yes
+    """
+
+    if input is not None and isinstance(input, dict) and len(input) == 1:
+
+        key = list(input.keys())[0]
+        return key == "Condition" or key.startswith("Fn::")
```
Comment (ShreyaGangishetty) [inline samtranslator/model/intrinsics.py]: If `!Sub` is given in the template does that get resolved to `Fn::Sub` at this point?

Comment (praneetap) [reply samtranslator/model/intrinsics.py]: sorry this change snuck in. Not using this new fn anywhere so will be getting rid of it

File: samtranslator/swagger/swagger.py [hunk 2]
```diff
@@ -824,16 +823,32 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
...
+                if not is_intrinsic_no_value(if_list[1]):
+                    self._add_custom_statement(if_list[1])
+                    if_true = self.resource_policy
+                self.resource_policy = original
+                if not is_intrinsic_no_value(if_list[2]):
+                    self._add_custom_statement(if_list[2])
+                    if_false = self.resource_policy
+                self.resource_policy = make_conditional(if_list[0], if_true, if_false)
+            else:
+                self._add_custom_statement(custom_statements)
```
Comment (keetonian) [inline samtranslator/swagger/swagger.py]: Why do we need to do this? Does it not work to just pass it straight through and let CFN resolve the if statement?

Comment (praneetap) [reply samtranslator/swagger/swagger.py]: The merge between global custom statements and local with `if` won't be handled.

Comment (keetonian) [reply samtranslator/swagger/swagger.py]: Could you add a test case then that tests this flow? I don't see a test case that has both local and global custom statements.

**Pull request merged**